plugins/markdown-extractor/src/parser.ts

```
@@ -0,0 +1,132 @@
+): Array<{ headers: string[]; rows: string[][] }> {
+  const tables: Array<{ headers: string[]; rows: string[][] }> = []
+  // More flexible regex pattern
+  const tableRegex =
+    /\|(.+?)\|[\r\n]+\|(?:[-:]+\|)+[\r\n]+((?:\|.+?\|[\r\n]+)+)/g
+  let match
+
+  if (options.debug) {
+    console.log('Content to parse:', content)
+  }
+
+  while ((match = tableRegex.exec(content)) !== null) {
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)